src/dtx/dtx_resync.c

```diff
@@ -141,7 +141,7 @@ dtx_is_leader(struct ds_pool *pool, struct dtx_resync_args *dra,
 	 *	data shard for EC object in the future.
 	 */
 	return ds_pool_check_dtx_leader(pool, &dre->dre_oid,
-					pool->sp_map_version);
+					pool->sp_map_version, false);
```

Comment:
@Nasf-Fan  If here (in dtx_resync) path set it as true to also check "oid->id_shard != leader_shard" in ds_pool_check_dtx_leader(), several DTX related test cases will fail, for example "daos_test -T -u 37" (DTX38: resync - lost whole redundancy groups), I have not found what's the reason.
So in this patch, I just added the "check_shard" parameter to make it pass test first.

Reply:
@Nasf-Fan could you look into it?